engine/src/cmd/collide2/Ice/IceTypes.h

```diff
@@ -90,31 +91,31 @@
 	typedef int	BOOL;						//!< Another boolean type.
 	#endif
 
-	//! Union of a float and a sdword
+	//! Union of a float and a ice_sdword
 	typedef union {
-		float	f;							//!< The float
-		sdword	d;							//!< The integer
+		float       f;                         //!< The float
+		ice_sdword  d;                         //!< The integer
 	}scell;
 
-	//! Union of a float and a udword
+	//! Union of a float and a ice_udword
 	typedef union {
-		float	f;							//!< The float
-		udword	d;							//!< The integer
+		float       f;                         //!< The float
+		ice_udword  d;                         //!< The integer
 	}ucell;
 
 	// Type ranges
-	#define	MAX_SBYTE				0x7f						//!< max possible sbyte value
-	#define	MIN_SBYTE				0x80						//!< min possible sbyte value
-	#define	MAX_UBYTE				0xff						//!< max possible ubyte value
-	#define	MIN_UBYTE				0x00						//!< min possible ubyte value
-	#define	MAX_SWORD				0x7fff						//!< max possible sword value
-	#define	MIN_SWORD				0x8000						//!< min possible sword value
-	#define	MAX_UWORD				0xffff						//!< max possible uword value
-	#define	MIN_UWORD				0x0000						//!< min possible uword value
-	#define	MAX_SDWORD				0x7fffffff					//!< max possible sdword value
-	#define	MIN_SDWORD				0x80000000					//!< min possible sdword value
-	#define	MAX_UDWORD				0xffffffff					//!< max possible udword value
-	#define	MIN_UDWORD				0x00000000					//!< min possible udword value
+	#define	MAX_SBYTE				0x7f						//!< max possible ice_sbyte value
+	#define	MIN_SBYTE				0x80						//!< min possible ice_sbyte value
+	#define	MAX_UBYTE				0xff						//!< max possible ice_ubyte value
+	#define	MIN_UBYTE				0x00						//!< min possible ice_ubyte value
+	#define	MAX_SWORD				0x7fff						//!< max possible ice_sword value
+	#define	MIN_SWORD				0x8000						//!< min possible ice_sword value
+	#define	MAX_UWORD				0xffff						//!< max possible ice_uword value
+	#define	MIN_UWORD				0x0000						//!< min possible ice_uword value
+	#define	MAX_SDWORD				0x7fffffff					//!< max possible ice_sdword value
+	#define	MIN_SDWORD				0x80000000					//!< min possible ice_sdword value
+	#define	MAX_UDWORD				0xffffffff					//!< max possible ice_udword value
+	#define	MIN_UDWORD				0x00000000					//!< min possible ice_udword value
```

Comment:
These probably should be cast to ensure they don't get accidentally converted to another value - e.g  an 8-bit signed max getting stored into a 16-bit value. I'd stick with the standard c11 types:
```
#define MAX_SBYTE int8_t(0x7F)
```
though if this is only for the ICE subsystem then using the ICE types would be okay too - `ice_sbyte(0x7F)`.

Reply:
If we used bit sized variables, we could probably use the system definitions and get rid of this. I'm assuming this matches some kind of system def.

Reply:
@royfalk yes, we could potentially declare these as C++ constants instead of C Macros; I'm not sure we need to right now though we might want to in the future; right now, I'm mostly wanting to make sure that the values don't get sign extended (https://en.wikipedia.org/wiki/Sign_extension), adding the types on each should at least make that visible to us via warnings.

Reply:
OK, I mostly switched to using the c++11 standard `INT32_MIN`, `INT32_MAX`, `UINT32_MAX`, etc. For the different width unsigned zeros, I do use `uint32_t(0x00000000)`, etc.

Again, though, do we really need these constants that are specific to ICE and OPCODE? Or can we just use `INT32_MAX` etc., directly?

Reply:
is there value added? If not, migrate.

Reply:
Migrated